products/billingbudget/api.yaml

```diff
@@ -177,7 +203,7 @@ objects:
                     - :CURRENT_SPEND
                     - :FORECASTED_SPEND
           - !ruby/object:Api::Type::NestedObject
-            name: allUpdatesRule
```

Comment:
Since this will be a breaking change, we should keep the old name in Terraform. Do you think it would be a significant improvement to use the new one? It's a lot of work to rename a nested block.

Doing the following should work to call it with the new name:

```
name: allUpdatesRule
api_name: notificationsRule
```

Reply:
v1 doesn't have the old name. It got renamed

Reply:
https://cloud.google.com/billing/docs/reference/budget/rest/v1/billingAccounts.budgets

Considering that field changes are fair game in beta APIs, our provider should be removing fields outside of major releases.

Reply:
I'm aware! Historically we've chosen to preserve the old name in these cases, though, because it's extremely disruptive to users with existing Terraform configs. I think we should do so here.

Reply:
Ah, I reread the suggestion. I'll retain the old name, tweak MM to add deprecated note and open an issue for f4

Reply:
I think we should avoid the deprecation entirely!

It's expected of providers that we allow both fields to be specified for at least one minor release before we remove the old field. Since it's mostly an aesthetic change to rename it, it isn't worth renaming it in Terraform since there's a decent amount of work to do allow users to specify either field at the same time, write the upgrade guide, and then ultimately remove the deprecated field.

We also found that `2.X` -> `3.0.0` was too large of an upgrade, so I'd like to avoid deprecations when possible to keep `4.0.0` a smaller upgrade.

Reply:
Ok, given that the backlog is huge i'll remove the deprecation.